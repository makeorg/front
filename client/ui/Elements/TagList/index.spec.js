import React from 'react';
import renderer from 'react-test-renderer';
import snapshotDiff from 'snapshot-diff';
import { TagList } from './index';

jest.mock('./style', () => ({
  TagListStyle: 'TagListStyle',
  TagButtonElementStyle: 'TagButtonElementStyle',
  TagListHeaderStyle: 'TagListHeaderStyle',
  TagElementUnderlinedStyle: 'TagElementUnderlinedStyle',
  TaglistWrapperStyle: 'TaglistWrapperStyle',
  TagListFooterStyle: 'TagListFooterStyle',
  TagLabelStyle: 'TagLabelStyle',
  CenterButtonStyle: 'CenterButtonStyle',
}));

jest.mock('Client/ui/Elements/Buttons/style', () => ({
  VoteIconStyle: 'VoteIconStyle',
}));

jest.mock('Client/ui/Elements/Form/SubmitButton/index', () => ({
  SubmitButton: 'SubmitButton',
}));

jest.mock('Client/ui/Svg/elements', () => ({
  SvgClose: 'SvgClose',
}));

jest.mock('../AccessibilityElements', () => ({
  ScreenReaderItemStyle: 'ScreenReaderItemStyle',
}));

const tagsArray = [
  { id: '1', label: 'foo', isSelected: true },
  { id: '2', label: 'bar', isSelected: false },
  { id: '3', label: 'baz', isSelected: false },
];

describe('TagList', () => {
  it('must match the snapshot with default Props', () => {
    const component = renderer.create(<TagList tags={tagsArray} />).toJSON();
    expect(component).toMatchSnapshot();
  });

  it('must match the diff snapshot between tagsSelected vs Default Props', () => {
    const tagsSelectedComponent = renderer
      .create(<TagList tags={tagsArray} tagsSelected={2} />)
      .toJSON();
    const defaultTagList = renderer
      .create(<TagList tags={tagsArray} />)
      .toJSON();
    expect(
      snapshotDiff(tagsSelectedComponent, defaultTagList)
    ).toMatchSnapshot();
  });
});
