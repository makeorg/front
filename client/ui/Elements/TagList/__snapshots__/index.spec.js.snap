// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagList must match the diff snapshot between tagsSelected vs Default Props 1`] = `
Snapshot Diff:
- First value
+ Second value

  <TaglistWrapperStyle>
    <TagListStyle />
    <TagListFooterStyle>
-     <CenterButtonStyle>
-       <ActiveButtonStyle>
-         consultation.proposal.see_proposals
-       </ActiveButtonStyle>
-     </CenterButtonStyle>
+     <span>
+       consultation.close
+     </span>
    </TagListFooterStyle>
    <TagListHeaderStyle>
      <span>
        consultation.tags.count
      </span>
-     <TagElementUnderlinedStyle
-       id="flush_tags_trigger"
-       onClick={[Function setReset]}
-     >
-       consultation.reset
-     </TagElementUnderlinedStyle>
    </TagListHeaderStyle>
  </TaglistWrapperStyle>
`;

exports[`TagList must match the snapshot with default Props 1`] = `
<TaglistWrapperStyle>
  <TagListStyle />
  <TagListFooterStyle>
    <span>
      consultation.close
    </span>
  </TagListFooterStyle>
  <TagListHeaderStyle>
    <span>
      consultation.tags.count
    </span>
  </TagListHeaderStyle>
</TaglistWrapperStyle>
`;
